Source/WebCore/Modules/mediasource/MediaSource.cpp

```diff
@@ -361,20 +361,24 @@ MediaTime MediaSource::duration() const
     // 1. If the readyState attribute is "closed" then return NaN and abort these steps.
     // 2. Return the current value of the attribute.
 
-    return m_private ? m_private->duration() : MediaTime::invalidTime();
+    if (RefPtr msp = protectedPrivate())
+        return msp->duration();
+    return MediaTime::invalidTime();
```

Comment:
Isn't duration a simple getter?

Reply:
it's not inline, so it won't be considered trivial